### PR TITLE
MGMT-13509: Introduce MCE operator (POC)

### PR DESCRIPTION
**Pull request by danielerez**
This PR introduces an MCE operator that can be deployed as part of a cluster installation. The operator is depended on the LVM operator, I.e. LVMS is also deployed during cluster installation.

The suggested flow here is just for demonstration purposes to illustrate the simplicity of using LVMS as a local persistent storage solution for an [effort](https://issues.redhat.com/browse/MGMT-12186) of a new MCE operator. So we have just the basic flow here without validations/tests/etc, as inspired by the original [PR](https://github.com/openshift/assisted-service/pull/4673) :)

Usage for example:
```
PATCH http://<service_url>/api/assisted-install/v2/clusters/<cluster_id>
{
   "olm_operators": [
       { "name": "mce" }
   ]
}
```

Note: LVMS is currently limited for SNO only, this [PR](https://github.com/openshift/assisted-service/pull/4936) is removing that limitation to support multi-node clusters as well.

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [x] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/controller/controllers/agentserviceconfig_controller.go [hunk 1]
```diff
@@ -1107,7 +1107,7 @@ func newAssistedCM(ctx context.Context, log logrus.FieldLogger, asc ASC) (client
 			"ISO_CACHE_DIR":          "/data/cache",
 
 			// from configmap
-			"AUTH_TYPE":                   "local",
+			"AUTH_TYPE":                   "none",
```
Comment (filanov) [inline internal/controller/controllers/agentserviceconfig_controller.go:1110]: is that temporary for testing?

Comment (danielerez) [reply internal/controller/controllers/agentserviceconfig_controller.go:1110]: Yeah, it's just a quick POC for testing purposes.

**Comment by eliorerz** [comment]
@danielerez Can you please also add subsystem test for it?

**Comment by danielerez** [comment]
> @danielerez Can you please also add subsystem test for it?

No need, it's already stale, I'll just close it:)

**Pull request closed without merging**